### PR TITLE
client: Support PKCS#11 3.0

### PR DESCRIPTION
**Pull request by ueno**
This exports `C_GetInterfaceList` and `C_GetInterface` from the client module.

### CODE REVIEW

File: p11-kit/client.c [hunk 1]
```diff
@@ -97,57 +97,176 @@ get_server_address (char **addressp)
...
+	}
+
+	state->wrapped.pInterfaceName = (char *)p11_interface_name;
+
+	state->wrapped.pFunctionList = (CK_FUNCTION_LIST_3_0 *)module;
+	module = NULL;
+
+	state->wrapped.flags = flags;
+
+	*interface = &state->wrapped;
```
Comment (ZoltanFridrich) [inline p11-kit/client.c:158]: I would put an assert on interface not being NULL

File: p11-kit/proxy.c [hunk 2]
```diff
@@ -2270,48 +2273,59 @@ get_interface_inlock(CK_INTERFACE **interface, const CK_VERSION *version, CK_FLA
...
+		module->version = *version;
+
+	state->wrapped.pInterfaceName = (char *)p11_interface_name;
+
+	state->wrapped.pFunctionList = module;
+	module = NULL;
+
+	state->wrapped.flags = flags;
+
+	*interface = &state->wrapped;
```
Comment (ZoltanFridrich) [inline p11-kit/proxy.c:2310]: assert that interface is not NULL

**Review by ZoltanFridrich — CHANGES_REQUESTED** [review]
Looks good, but I have some comments.

File: p11-kit/client.c [hunk 3]
```diff
@@ -97,57 +97,176 @@ get_server_address (char **addressp)
...
 	}
 
-	p11_unlock ();
+	p11_library_init_once ();
+	p11_lock ();
 
-	free (address);
+	if (pVersion &&
+	    memcmp (pVersion, &version_three, sizeof(*pVersion)) == 0)
+		pVersion = NULL;
```
[the remark this answers is not in the document]
Comment (ueno) [reply p11-kit/client.c]: Indeed, that's a good point; I've removed the special case and made `get_interface_inlock` always expects non-NULL `version`.

**Review by ZoltanFridrich — APPROVED** [review]

**Pull request merged**